app/Http/Controllers/Product/ProductController.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace App\Http\Controllers\Product;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Models\Bid;
+use App\Models\User_Bid;
+use DB;
+use App\Repositories\Bid\BidRepositoryInterface;
+use App\Repositories\User_Bid\UserBidRepositoryInterface;
+use Illuminate\Support\Facades\Config;
+use Carbon\Carbon;
+class ProductController extends Controller
+{
+	protected $bidRepository;
+    protected $user_bidRepository;
+    protected $time_now;
+    protected $bid_amount_step;
+    protected $bid_type_manual;
+    protected $bid_type_auto;
+	public function __construct(BidRepositoryInterface $bidRepository,UserBidRepositoryInterface $user_bidRepository) {
+		$this->bidRepository = $bidRepository;
+        $this->user_bidRepository = $user_bidRepository;
+        $this->time_now = Carbon::now();
+        $this->bid_amount_step = Config::get('constants.bid_amount_step');
+        $this->bid_type_manual = Config::get('constants.bid_amount_type.manual');
+        $this->bid_type_auto = Config::get('constants.bid_amount_type.auto');
+	}
+
+    /**
+     * @param $id
+     * @return Bid
+     */
+    public function getBid($id) {
+        $bid = $this->bidRepository->find($id);
+        return view('bid.bid_current',['bid'=>$bid]);
+    }
+
+    /**
+     * @param $request{user_id, real_bid_amound}
+     * @return $current_price
+     */
+    public function postBid(Request $request, $id) {
+        $bid = $this->bidRepository->find($id);
+        
+        if($bid->status == Config::get('constants.bid_status.begining')){
+            if($request->real_bid_amount <=  ($bid->current_price)){
+                return false;
+            }elseif ($request->real_bid_amount >= $bid->cost_sell) {
+                $this->amountCostSell($bid, $request, $id);
+            }else{
+                if($request->real_bid_amount < $bid->current_highest_price){
+                    $this->amountLessHightest($bid, $request, $id);
+                }elseif($request->real_bid_amount == $bid->current_highest_price){
+                    $this->amountEqualHightest($bid, $request, $id);
+                }else{
+                    $this->amountBetterHightest($bid, $request, $id);
+                }
+            }
+            
+        }elseif ($bid->status == Config::get('constants.bid_status.no_user_bid')) {
+
+            $this->amountNoUserBid($bid, $request, $id);
+        }else
+            return false;
+        $respont = array(
+                'current_highest_bidder_id'=> $bid->current_highest_bidder_id,
+                'current_highest_bidder_name'=>$bid->current_highest_bidder_name,
```

Comment:
Tạm chưa nhắc đến các hàm tính toán bên dưới anh chỉ review hàm bid thì thấy: 
Case trả giá lần đầu tiên a k thấy, ngoài ra khi so sánh a thấy thiếu bước nhảy giá.